docs/source/geo_def.rst

```diff
@@ -1,84 +1,69 @@
 Geometry definitions
 ====================
-The module **pyresample.geometry** contains classes for describing different kinds of types
-of remote sensing data geometries. The use of the different classes is described below.
 
-Remarks
--------
+The :mod:`pyresample.geometry` module contains classes for describing different
+geographic areas using a mesh of points or pixels. Some classes represent
+geographic areas made of of evenly spaced/sized pixels, others handle the cases
+where the region is described by non-uniform pixels. The best object for describing a
+region depends on the use case and the information known about it. The different
+classes available in pyresample are described below.
 
-All longitudes and latitudes provided to **pyresample.geometry** must be in degrees.
-Longitudes must additionally be in the [-180;+180[ validity range.
+Note that all longitudes and latitudes provided to :mod:`pyresample.geometry`
+classes must be in degrees. Additionally, longitudes must be in the
+[-180;+180[ validity range.
 
-As of version 1.1.1, the **pyresample.geometry** contructors will check the range of
-longitude values, send a warning if some of them fall outside validity range,
-and automatically correct the invalid values into [-180;+180[.
+.. versionchanged:: 1.8.0
 
-Use function **utils.wrap_longitudes** for wrapping longitudes yourself.
+    Geometry objects no longer check the validity of the provided longitude
+    and latitude coordinates to improve performance. Longitude arrays are
+    expected to be between -180 and 180 degrees, latitude -90 to 90 degrees.
+    This also applies to all geometry definitions that are provided longitude
+    and latitude arrays on initialization. Use
+    :func:`~pyresample.utils.check_and_wrap` to preprocess your arrays.
 
 AreaDefinition
 --------------
 
-The cartographic definition of grid areas used by Pyresample is contained in an
-object of type AreaDefintion. The following arguments are needed to initialize
-an area:
+An :class:`~pyresample.geometry.AreaDefinition`, or ``area``, is the primary
+way of specifying a uniformly spaced geographic region in pyresample. It is
+also one of the only geometry objects that understands geographic projects.
+Areas use the :doc:`PROJ.4 <proj4:index>` method for describing projected
+coordinate reference systems (CRS). If the projection for an area is not
+described by longitude/latitude coordinates then it is typically described
+in X/Y coordinates in meters. See the PROJ.4 documentation for more
+information on projections and coordinate reference systems.
 
-* **area_id** ID of area
-* **name**: Description
+The following arguments are needed to initialize an area:
+
+* **area_id**: ID of area
+* **description**: Description
 * **proj_id**: ID of projection (being deprecated)
```

Comment:
Do you want a fully incompatible class interface? Right now the positional arguments still mean the same thing even though they have different names. I can add a warning if the `proj_id` is not None or doesn't equal area_id, but not sure that helps much since code would still break in the future.

Reply:
By removing `get_area_def` we are breaking backwards compatibility, so this will be released as 2.x.x anyway...

Reply:
I didn't remove `get_area_def`, only talked about it on slack. I was going to deprecate it but it's recent functionality of returning an AreaDefinition *or* a DynamicAreaDefinition is not available anywhere else. If `from_params` is renamed to `get_area_def` and provides the same interface I don't think we need to go all the way to 2.x.x.

I'd prefer to release this functionality as a 1.x release and have a real 2.x design process (with issues marked for 2.0 and really considering new interface opportunities).

Reply:
We leave this be then.